packages/allure-cypress/src/reporter.ts

```diff
@@ -50,13 +51,42 @@ export const allureCypress = (on: Cypress.PluginEvents, config?: AllureCypressCo
       resultsDir: config?.resultsDir || "./allure-results",
     }),
   });
+  const currentTestsByAbsolutePath = new Map<string, [AllureTest, number][]>();
+  // need to keep the variable here because we recieve end-message and finish the current test separately
   const currentSteps: AllureStep[] = [];
 
+  on("after:spec", (spec, results) => {
```

Comment:
This hook doesn't get called in interactive mode even with `experimentalInteractiveRunEvents` set to `true` (at least, that's what I've observed on my machine).

However, we may benefit from the fact that Cypress doesn't record videos in interactive mode. That means we don't need to wait for `after:spec` to close the test. If we pass the result of `Cypress.config("isInteractive")` (see [here](https://glebbahmutov.com/blog/cypress-tips-and-tricks/#interactive-and-headed-mode)) to the `allureReportTest` task, we will be able to decide on whether to end the test in the task or delay the ending until `"after:spec"` is called.

Also, unlike `task`, `after:spec` handlers [can't be composed](https://github.com/cypress-io/cypress/issues/5240). The user may define their own `after:spec` handler that either overrides ours or is overridden by us depending on where they put it in their code.

We may compensate for that by factoring out the handler to a function and exporting it for the user to compose it (in addition to attaching it to `after:spec` in our code):

```js
const { allureCypress, attachVideos } = require("allure-cypress/reporter");
module.exports = {
  e2e: {
    video: true,
    setupNodeEvents: (on, config) => {
      allureCypress(on, {});
      on("after:spec", (s, r) => {
        /* the user's code */
        /* ... */

        attachVideos(s, r);
      });
      return config;
    },
  },
};
```

Not very friendly but the best we can do.

Reply:
You're right. Will check, how we can solve the problem

Reply:
Done